src/common.ts

```diff
@@ -61,3 +63,36 @@ export function shamirRecovery(x1: bigint, x2: bigint, y1: bigint, y2: bigint):
 export function calculateIdentityCommitment(identitySecret: bigint) {
   return poseidon([identitySecret])
 }
+
+export function isValidUrl(str: string): boolean {
+  try {
+    new URL(str)
+    return true
+  } catch (_) {
+    return false
+  }
+}
+
+export async function checkFileExistsOnWeb(url: string): Promise<boolean> {
+  try {
+    await axios.head(url)
+    return true
+  } catch (error) {
+    return false
+  }
+}
+
+export async function checkFileExists(path: string): Promise<boolean> {
+  if (isValidUrl(path)) {
+    return await checkFileExistsOnWeb(path)
+  } else {
+
+    if (typeof window !== 'undefined' && typeof window.document !== 'undefined') {
+      throw new Error(
+        'not allowed to read local files from browser',
+      )
+    }
+
```

Comment:
Can we move `import fs from 'fs'` and use a dynamic import here? Just wanted to make sure `fs` is imported only in Nodejs.
```suggestion

    // It's in NodeJS so we can safely import and use `fs`
    const fs = await import("fs");
```

To use dynamic imports, we need to change the `module` item in `tsconfig.json` to use a newer ES Module.

Change "ES6" to "ES2020"

```
"module": "ES2020"
```